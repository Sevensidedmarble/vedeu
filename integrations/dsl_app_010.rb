#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    log '/tmp/vedeu_views_dsl.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename:   '/tmp/dsl_app_010.out',
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: '/tmp/dsl_app_010.json'),
                # Vedeu::Renderers::HTML.new(filename: '/tmp/dsl_app_010.html'),
                # Vedeu::Renderers::Text.new(filename: '/tmp/dsl_app_010.txt'),
              ]
    run_once!
    standalone!
  end

  load './support/test_interface.rb'

  Vedeu.render do
    view(:test_interface) do
      line do
        streams do
          stream 'Multiple ', { foreground: '#00ff00' }
          stream '`stream` ', { foreground: '#007fff' }
          stream 'within a ', { foreground: '#00ff00' }
        end
      end
      line do
        streams do
          stream '`streams` ', { foreground: '#007fff' }
          stream 'within a ',  { foreground: '#00ff00' }
          stream '`line`',     { foreground: '#007fff' }
          stream '.',          { foreground: '#00ff00' }
        end
      end
    end
  end

  def self.actual
    File.read('/tmp/dsl_app_010.out')
  end

  def self.expected
    File.read('./expected/dsl_app_010.out')
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

DSLApp.start

if DSLApp.expected == DSLApp.actual
  puts "#{__FILE__} \e[32mPassed.\e[39m"
else
  puts "#{__FILE__} \e[31mFailed.\e[39m"
end
