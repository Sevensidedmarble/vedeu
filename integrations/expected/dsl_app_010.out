[0m[0m[2;2H[39m[49m(0l(B[2;3H(0q(B[2;4H [2;5HT[2;6He[2;7Hs[2;8Ht[2;9H [2;10H(0q(B[2;11H(0q(B[2;12H(0q(B[2;13H(0q(B[2;14H(0q(B[2;15H(0q(B[2;16H(0q(B[2;17H(0q(B[2;18H(0q(B[2;19H(0q(B[2;20H(0q(B[2;21H(0q(B[2;22H(0q(B[2;23H(0q(B[2;24H(0q(B[2;25H(0q(B[2;26H(0q(B[2;27H(0q(B[2;28H(0q(B[2;29H(0q(B[2;30H(0q(B[2;31H(0q(B[2;32H(0k(B[0m[3;2H[39m[49m(0x(B[3;3H[38;5;46m[49mM[3;4Hu[3;5Hl[3;6Ht[3;7Hi[3;8Hp[3;9Hl[3;10He[3;11H [3;12H[38;5;33m[49m`[3;13Hs[3;14Ht[3;15Hr[3;16He[3;17Ha[3;18Hm[3;19H`[3;20H [3;21H[38;5;46m[49mw[3;22Hi[3;23Ht[3;24Hh[3;25Hi[3;26Hn[3;27H [3;28Ha[3;29H [3;30H[39m[49m [3;31H [3;32H(0x(B[0m[4;2H[39m[49m(0x(B[4;3H[38;5;33m[49m`[4;4Hs[4;5Ht[4;6Hr[4;7He[4;8Ha[4;9Hm[4;10Hs[4;11H`[4;12H [4;13H[38;5;46m[49mw[4;14Hi[4;15Ht[4;16Hh[4;17Hi[4;18Hn[4;19H [4;20Ha[4;21H [4;22H[38;5;33m[49m`[4;23Hl[4;24Hi[4;25Hn[4;26He[4;27H`[4;28H[38;5;46m[49m.[4;29H[39m[49m [4;30H [4;31H [4;32H(0x(B[0m[5;2H[39m[49m(0x(B[5;3H [5;4H [5;5H [5;6H [5;7H [5;8H [5;9H [5;10H [5;11H [5;12H [5;13H [5;14H [5;15H [5;16H [5;17H [5;18H [5;19H [5;20H [5;21H [5;22H [5;23H [5;24H [5;25H [5;26H [5;27H [5;28H [5;29H [5;30H [5;31H [5;32H(0x(B[0m[6;2H[39m[49m(0m(B[6;3H(0q(B[6;4H(0q(B[6;5H(0q(B[6;6H(0q(B[6;7H(0q(B[6;8H(0q(B[6;9H(0q(B[6;10H(0q(B[6;11H(0q(B[6;12H(0q(B[6;13H(0q(B[6;14H(0q(B[6;15H(0q(B[6;16H(0q(B[6;17H(0q(B[6;18H(0q(B[6;19H(0q(B[6;20H(0q(B[6;21H(0q(B[6;22H(0q(B[6;23H(0q(B[6;24H(0q(B[6;25H(0q(B[6;26H(0q(B[6;27H(0q(B[6;28H(0q(B[6;29H(0q(B[6;30H(0q(B[6;31H(0q(B[6;32H(0j(B[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m[0m